astronomer/providers/amazon/aws/xcom_backends/s3.py

```diff
@@ -0,0 +1,66 @@
+import json
+import os
+import uuid
+from typing import Any, Union
+
+import pandas as pd
+from airflow.models.xcom import BaseXCom
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+
+
+class S3XComBackend(BaseXCom):
+    """
+    Custom XCom persistence class extends base to support various datatypes.
+    To use this XCom Backend, add the environment variable `AIRFLOW__CORE__XCOM_BACKEND`
+    to your environment and set it to
+    `astronomer.providers.amazon.aws.xcom_backends.s3.S3XComBackend`
+    """
+
+    PREFIX = os.getenv("PREFIX", "s3_xcom_")
+    AWS_CONN_ID = os.getenv("CONNECTION_NAME", "aws_default")
+    BUCKET_NAME = os.getenv("XCOM_BACKEND_BUCKET_NAME", "airflow_xcom_backend_default_bucket")
+
+    @staticmethod
+    def write_and_upload_value(value: Any) -> str:
+        """Convert to string and upload to GCS"""
+        key_str = S3XComBackend.PREFIX + str(uuid.uuid4())
+        hook = S3Hook(aws_conn_id=S3XComBackend.AWS_CONN_ID)
+        if isinstance(value, list):
+            value = str(value)
+        elif isinstance(value, dict):
+            value = json.dumps(value)
+        elif isinstance(value, pd.DataFrame):
+            value = value.to_json()
+        hook.load_string(bucket_name=S3XComBackend.BUCKET_NAME, key=key_str, string_data=value)
+        return key_str
+
+    @staticmethod
+    def read_value(filepath: str) -> Union[str, bytes]:
+        """Download the file from S3"""
+        # Here we download the file from S3
+        hook = S3Hook(aws_conn_id=S3XComBackend.AWS_CONN_ID)
+        data = hook.download_file(bucket_name=S3XComBackend.BUCKET_NAME, key=filepath)
```

Comment:
what happened if the file downloading is failed or file is not present in the bucket ?

Reply:
it automatically throws an airflow exception with details, I can add an exception and give the details if needed but it will be a duplicate.